iothub/device/src/Authentication/IotHubConnectionCredentials.cs

```diff
@@ -331,6 +318,12 @@ private void Validate()
                 throw new FormatException("Device Id cannot be null or white space.");
             }
 
+            // Module Id
+            if (ModuleId != null && string.IsNullOrWhiteSpace(ModuleId))
+            {
+                throw new FormatException("Module Id cannot be white space.");
```

Comment:
module Id can be null but cannot be white space

Reply:
// A module Id is not required to provide, but if they do it must be a valid string.